src/flexbox.rs

```diff
@@ -206,31 +203,29 @@ impl Forest {
 
     /// Compute constants that can be reused during the flexbox algorithm.
     #[inline]
-    fn compute_constants(
-        &self,
-        node: NodeId,
-        node_size: Size<Option<f32>>,
-        parent_size: Size<Option<f32>>,
-    ) -> AlgoConstants {
+    fn compute_constants(&self, node: NodeId, node_size: Size<f32>, parent_size: Size<f32>) -> AlgoConstants {
         let dir = self.nodes[node].style.flex_direction;
         let is_row = dir.is_row();
         let is_column = dir.is_column();
         let is_wrap_reverse = self.nodes[node].style.flex_wrap == FlexWrap::WrapReverse;
 
-        let margin = self.nodes[node].style.margin.map(|n| n.resolve(parent_size.width).unwrap_or(0.0));
-        let padding = self.nodes[node].style.padding.map(|n| n.resolve(parent_size.width).unwrap_or(0.0));
-        let border = self.nodes[node].style.border.map(|n| n.resolve(parent_size.width).unwrap_or(0.0));
+        let margin =
+            self.nodes[node].style.margin.map(|n| n.map(|d| d.resolve(parent_size.width)).unwrap_or(Some(0.0)));
+        let padding =
+            self.nodes[node].style.padding.map(|n| n.map(|d| d.resolve(parent_size.width)).unwrap_or(Some(0.0)));
+        let border =
+            self.nodes[node].style.border.map(|n| n.map(|d| d.resolve(parent_size.width)).unwrap_or(Some(0.0)));
```

Comment:
I'm very uncertain if this is the correct logic. Still have more work to do before everything compiles and tests can start to yell at me.